src/Symfony/Component/Workflow/WorkflowInterface.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow;
+
+use Symfony\Component\Workflow\Exception\LogicException;
+use Symfony\Component\Workflow\MarkingStore\MarkingStoreInterface;
+
+/**
+ * @author Amrouche Hamza <hamza.simperfit@gmail.com>
+ */
+interface WorkflowInterface
+{
+    /**
+     * Returns the object's Marking.
+     *
+     * @param object $subject A subject
+     *
+     * @return Marking The Marking
+     *
+     * @throws LogicException
+     */
+    public function getMarking($subject);
+
+    /**
+     * Returns true if the transition is enabled.
+     *
+     * @param object $subject        A subject
+     * @param string $transitionName A transition
+     *
+     * @return bool true if the transition is enabled
+     */
+    public function can($subject, $transitionName);
```

Comment:
Hm, if getMarking can throw an exception, then can and apply may throw an exception too